app/sagas/room.js

```diff
@@ -14,18 +14,43 @@ import I18n from '../i18n';
 import { showErrorAlert } from '../utils/info';
 import { LISTENER } from '../containers/Toast';
 
-const watchUserTyping = function* watchUserTyping({ rid, status }) {
+const timeoutTimer = 10000;
+const renewTimer = 4000;
+
+const activityRenews = {};
+
+const watchUserActivity = function* watchUserActivity({
+	type, rid, status, options = {}
+}) {
+	const id = options?.tmid ? options.tmid : rid;
+
+	if (id in activityRenews && status === true) {
+		yield delay(2000);
+	}
```

Comment:
Is it a correct way of limiting server requests?

Reply:
@diegolmello I think only you or your team can answer here, I don't have that knowledge